samples/ClassroomLabs/Modules/Library/Tools/LabCreator.ps1

```diff
@@ -267,20 +271,42 @@ $labs = Import-Csv -Path $CsvConfigFile
 Write-Verbose ($labs | Format-Table | Out-String)
 
 $labs | ForEach-Object {
-    if ($_.Emails) {
+
+    # Validate that the name is good, before we start creating labs
+    if (-not ($_.LabName -match "^[a-zA-Z0-9_, '`"!|-]*$")) {
+        Write-Error "Lab Name '$($_.LabName)' can't contain special characters..."
+    }
+
+    if ((Get-Member -InputObject $_ -Name 'Emails') -and ($_.Emails)) {
         $_.Emails = ($_.Emails.Split(';')).Trim()
     }
 
     if (Get-Member -InputObject $_ -Name 'GpuDriverEnabled') {
-        $_.GpuDriverEnabled = [System.Convert]::ToBoolean($_.GpuDriverEnabled)
+        if ($_.GpuDriverEnabled) {
+            $_.GpuDriverEnabled = [System.Convert]::ToBoolean($_.GpuDriverEnabled)
+        }
+        else {
+            $_.GpuDriverEnabled = $false
+        }
+    }
+    else {
+        Add-Member -InputObject $_ -MemberType NoteProperty -Name "GpuDriverEnabled" -Value $false
     }
 
     if (Get-Member -InputObject $_ -Name 'LinuxRdp') {
-        $_.LinuxRdp = [System.Convert]::ToBoolean($_.LinuxRdp)
+        if ($_.LinuxRdp) {
+            $_.LinuxRdp = [System.Convert]::ToBoolean($_.LinuxRdp)
+        }
+        else {
+            $_.LinuxRdp = $false
+        }
+    }
+    else {
+        Add-Member -InputObject $_ -MemberType NoteProperty -Name "LinuxRdp" -Value $false
     }
 
     $_.SharedPassword = [System.Convert]::ToBoolean($_.SharedPassword)
-    if ($_.Schedules) {
+    if ((Get-Member -InputObject $_ -Name 'Schedules') -and ($_.Schedules)) {
```

Comment:
Fixing to make sure the script can run if the Schedules entry is missing (not present in the CSV file)